src/websocket_client.rs

```diff
@@ -22,102 +23,167 @@ pub struct WebsocketService {
+                if let (Some(association_id), Some(candidate_id)) = (
+                    get_uuid_in_path(req.uri().path(), 2),
+                    get_uuid_in_path(req.uri().path(), 3),
+                ) {
+                    if let Some(association) = jet_associations.get_mut(&association_id) {
+                        if association.version() == JET_VERSION_V2 {
+                            let res = process_req(&req);
 
-            if let Some(header) = req.headers().get("upgrade") {
-                if header.to_str().ok().filter(|s| s == &"websocket").is_some() {
-                    if let Ok(mut jet_associations) = self.jet_associations.lock() {
-                        if let (Some(association_id), Some(candidate_id)) = (get_uuid_in_path(req.uri().path(), 2), get_uuid_in_path(req.uri().path(), 3)) {
-                            if let Some(association) = jet_associations.get_mut(&association_id) {
-                                if association.version() == JET_VERSION_V2 {
-                                    let res = process_req(&req);
-
-                                    let jet_associations_clone = self.jet_associations.clone();
-                                    let self_clone = self.clone();
-                                    let fut = req.into_body().on_upgrade().map(move |upgraded| {
-                                        if let Ok(mut jet_assc) = jet_associations_clone.lock() {
-                                            if let Some(assc) = jet_assc.get_mut(&association_id) {
-                                                if let Some(candidate) = assc.get_candidate_mut(candidate_id) {
-                                                    if (candidate.transport_type() == TransportType::Ws || candidate.transport_type() == TransportType::Wss) && candidate.state() == CandidateState::Accepted {
-                                                        candidate.set_client_transport(JetTransport::Ws(WsTransport::new_http(upgraded, client_addr)));
+                            let executor_handle_clone = executor_handle.clone();
+                            let fut = req
+                                .into_body()
+                                .on_upgrade()
+                                .map(move |upgraded| {
+                                    if let Ok(mut jet_assc) = jet_associations_clone.lock() {
+                                        if let Some(assc) = jet_assc.get_mut(&association_id) {
+                                            if let Some(candidate) = assc.get_candidate_mut(candidate_id) {
+                                                if (candidate.transport_type() == TransportType::Ws || candidate.transport_type() == TransportType::Wss)
+                                                    && candidate.state() == CandidateState::Accepted {
+                                                    {
+                                                        let server_transport = candidate
```

Comment:
Here the same thing happens as in jet_client.rs: take transport from the candidate, set the client's nb_bytes_read/written, and create the proxy.